### DISCUSSION TITLE
Internal RPC error

### DISCUSSION THREAD

**Discussion by adijdhv** [Web3Auth/Web3Auth#419] [Help]
Hello I've been searching for solution for this error. I'm Getting Internal RPC error while interacting with Smart Contract as :- 
`Error: Internal JSON-RPC error.
    at getEthJsonRpcError (errors.ts:188:1)
    at Object.internal (errors.ts:52:1)
    at provider.sendAsync (INetworkController.ts:125:1)
    at async provider.request (INetworkController.ts:149:1)`  

Digging further I Got to know I'm getting provider as Follows :-
`Proxy {}
[[Handler]]: Object
get: (_, name) => {…}
set: (_, name, value) => {…}
length: 3
name: "set"
arguments: [Exception: TypeError: 'caller', 'callee', and 'arguments' properties may not be accessed on strict mode functions or the arguments objects for calls to them at Function.invokeGetter (<anonymous>:3:28)]
caller: [Exception: TypeError: 'caller', 'callee', and 'arguments' properties may not be accessed on strict mode functions or the arguments objects for calls to them at Function.invokeGetter (<anonymous>:3:28)]
[[FunctionLocation]]: createEventEmitterProxy.ts:45
[[Prototype]]: ƒ ()
[[Scopes]]: Scopes[4]
[[Prototype]]: Object
[[Target]]: Object
[[Prototype]]: Object
[[IsRevoked]]: false`

arguments and Caller are throwing TypeError. Any Solution will be helpful. Thank you.

**Comment by himanshuchawla009** [comment]
plz check if any of the network call it failing.

**Comment by yashovardhan** [comment]
@adijdhv were you able to figure this out?

**Reply by Yajassardana** [reply]
Hey @yashovardhan, I'm facing the same issue.
I'm calling "approve" on an ERC20 contract whose implementation is done using an underlying proxy contract.

Now this call is working if I login using metamask on web3auth. However, if I login using any other method like gmail, I get the same error as mentioned in the thread.

My hunch is that in the former case, metamask is taking care of the gas estimations and working fine, where in the latter, web3auth is struggling with the gas estimations to call "approve" which is defined in the underlying proxy contract. If I call **read** functions like `balanceOf`, these work even with gmail login, as the do not involve any gas.

I'm running on a crunched timeline for my project, any help on this would be super helpful.

**Comment by oscar88leonardo** [comment]
hello, does it work for you when you log in with metamask (MM)  via web3auth and then interact with your contract ?. 

 I am getting the same error, it works fine when i log in with MM, but same error when i log in with google via web3auth. 

i think it could be something related with web3auth configuration to send transaction and set the gaslimit and the transaction parameters that MM handle when login with the MM wallet via web3auth.

please let me know if you find a solution

**Reply by tse-lao** [reply]
I encounter the same issue, anyone found a solution yet?
